### PR TITLE
ci: add script to support run group tests on ci

### PR DESCRIPTION
**Pull request by purelind**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/8667

#### Tests 

 - No code

##### Will it cause performance regression or break compatibility?
None
##### Do you need to update user documentation, design documentation or monitoring documentation?
None
### Release note 

```release-note
None
```

### CODE REVIEW

File: dm/tests/download-compatibility-test-binaries.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+# Copyright 2021 PingCAP, Inc.
```
Comment (amyangfei) [inline dm/tests/download-compatibility-test-binaries.sh:2]:
```suggestion
# Copyright 2023 PingCAP, Inc.
```

File: tests/integration_tests/run_group.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+set -eo pipefail
+
+CUR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+
+sink_type=$1
+group=$2
+
+# Define groups
+# Note: If new group is added, the group name must also be added to CI
+# https://github.com/PingCAP-QE/ci/blob/main/pipelines/tikv/migration/latest/pull_integration_test.groovy
```
Comment (amyangfei) [inline tests/integration_tests/run_group.sh:12]: Is this link correct?

File: tests/integration_tests/README.md [hunk 3]
```diff
@@ -120,5 +120,9 @@ Some useful tips:
 
 ## Writing new tests
 
-New integration tests can be written as shell scripts in `tests/integration_tests/TEST_NAME/run.sh`. The script should
+1. New integration tests can be written as shell scripts in `tests/integration_tests/TEST_NAME/run.sh`. The script should
 exit with a nonzero error code on failure.
+
+2. Add TEST_NAME to existing group in [run_group.sh](./run_group.sh), or add a new group for it.
+
+3. If you add a new group, the name of the new group must be added to [CI](https://github.com/PingCAP-QE/ci/blob/main/pipelines/pingcap/tiflow/latest/pull_cdc_integration_mysql_test.groovy).
```
Comment (amyangfei) [inline tests/integration_tests/README.md:128]: This link can't be open

File: tests/integration_tests/run_group.sh (hunk 2 above)
Comment (purelind) [reply tests/integration_tests/run_group.sh:12]: i will check and fix them later

File: tests/integration_tests/README.md (hunk 3 above)
Comment (purelind) [reply tests/integration_tests/README.md]: fixed

File: tests/integration_tests/run_group.sh (hunk 2 above)
Comment (purelind) [reply tests/integration_tests/run_group.sh]: update with correct link.

**Review by amyangfei — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -552,3 +559,7 @@ engine_unit_test_in_verify_ci: check_failpoint_ctl tools/bin/gotestsum tools/bin
 	tools/bin/gocov convert "$(ENGINE_TEST_DIR)/cov.unit_test.out" | tools/bin/gocov-xml > engine-coverage.xml
 	$(FAILPOINT_DISABLE)
 	@bash <(curl -s https://codecov.io/bash) -F engine -f $(ENGINE_TEST_DIR)/cov.unit_test.out -t $(TICDC_CODECOV_TOKEN)
+
+prepare_test_binaries:
+	cd scripts && ./download-integration-test-binaries.sh master && cd ..
```
Comment (wuhuizuo) [inline Makefile:564]: The constant `master` will broke the next release branch's CI results.

**Pull request merged**